community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/commands/expressions/ToIntFunction.scala

```diff
@@ -39,12 +39,17 @@ case class ToIntFunction(a: Expression) extends NullInNullOutExpression(a) {
       v.longValue()
     case v: String =>
       try {
-        val d = BigDecimal(v)
-        if (d <= Long.MaxValue && d >= Long.MinValue) d.toLong
-        else throw new CypherTypeException(s"integer, $v, is too large")
+        java.lang.Long.parseLong(v)
       } catch {
-        case e: NumberFormatException =>
-          null
+        case e: Exception =>
+        try {
+          val d = BigDecimal(v)
```

Comment:
For what values `v` will `BigDecimal(v)` + range check succeed for which `parseLong` will fail?